### DISCUSSION TITLE
Isn't the failed login log too verbose for an INFO level?

### DISCUSSION THREAD

**Discussion by wsw70** [authelia/authelia#3357] [Ideas]
Upon a failed login, I get in the logs

```
time="2022-05-13T15:58:28+02:00" level=info msg="Access to https://whoami.example.eu/ (method GET) is not authorized to user <anonymous>, responding with status code 302 with location redirect to https://auth.example.eu?rd=https%3A%2F%2Fwhoami.example.eu%2F&rm=GET" method=GET path=/api/verify remote_ip=192.168.10.1
time="2022-05-13T15:59:21+02:00" level=error msg="Unsuccessful 1FA authentication attempt by user 'XXX'" method=POST path=/api/firstfactor remote_ip=192.168.10.1 stack="github.com/authelia/authelia/v4/internal/handlers/response.go:237           markAuthenticationAttempt\ngithub.com/authelia/authelia/v4/internal/handlers/handler_firstfactor.go:61 FirstFactorPOST.func1\ngithub.com/authelia/authelia/v4/internal/middlewares/authelia_context.go:53 AutheliaMiddleware.func1.1\ngithub.com/authelia/authelia/v4/internal/middlewares/headers.go:25          SecurityHeadersCSPNone.func1\ngithub.com/authelia/authelia/v4/internal/middlewares/headers.go:35          SecurityHeadersNoStore.func1\ngithub.com/authelia/authelia/v4/internal/middlewares/headers.go:16          SecurityHeaders.func1\ngithub.com/fasthttp/router@v1.4.9/router.go:414                             (*Router).Handler\ngithub.com/valyala/fasthttp@v1.36.0/http.go:153                             (*Response).StatusCode\ngithub.com/valyala/fasthttp@v1.36.0/server.go:2338                          (*Server).serveConn\ngithub.com/valyala/fasthttp@v1.36.0/workerpool.go:224                       (*workerPool).workerFunc\ngithub.com/valyala/fasthttp@v1.36.0/workerpool.go:196                       (*workerPool).getCh.func1\nruntime/asm_amd64.s:1571
                  goexit"
```

I feel that this is too much information for an `INFO` log level. I believe it is appropriate for `DEBUG` and for `INFO` it should just be the above, without the `stack`:

```
time="2022-05-13T15:58:28+02:00" level=info msg="Access to https://whoami.example.eu/ (method GET) is not authorized to user <anonymous>, responding with status code 302 with location redirect to https://auth.example.eu?rd=https%3A%2F%2Fwhoami.example.eu%2F&rm=GET" method=GET path=/api/verify remote_ip=192.168.10.1
time="2022-05-13T15:59:21+02:00" level=error msg="Unsuccessful 1FA authentication attempt by user 'XXX'" method=POST path=/api/firstfactor remote_ip=192.168.10.1
```

**Comment by james-d-elliott** [comment]
Only one of those is generated on a failed login, it's the second line. The first line occurs on a redirection. As to if that first line is too verbose... maybe?

**Comment by MrJake222** [comment]
I agree, why the `stack` is printed. What does it even mean? Why there are github links there?

**Reply by james-d-elliott** [reply]
They are not links, they are module names. Stacks are printed for all errors to show the actual cause. It being too verbose is very subjective until you consider general security principles. The failed login is critically important to know about as it may be an indication of brute force attempts. Without good cause we're unlikely to change it.

**Reply by MrJake222** [reply]
But the request here isn't removing the failed login from info loglevel completely. Just printing it without stack trace. Stack trace is just too verbose for info level.

**Reply by james-d-elliott** [reply]
This is completely subjective. The log level determines what log levels are shown, not what's shown in them. This shouldn't be changed.

Adjusting this behavior will drastically affect the support burden for both the maintainers and the users alike. Why? Some errors obviously may not occur frequently and users may not know the underlying cause making it hard for them to reproduce. When it comes to finding a stack trace for those errors it's generally unreasonable to have a person log at a debug or trace level until it occurs, so we can dive deeper into the cause of the issue.

I may keep in mind adding an option for future logging reworks to exclude all stack traces, but all users who use it will likely be refused support.

There are however no compelling reasons to change this (or none have been presented), only compelling reasons to keep it as is.
